plugin-cgroupv2/src/k8s/token.rs

```diff
@@ -55,70 +55,65 @@ impl Token {
     /// Returns the current value of the token if it is still alive or
     /// it's refreshed value if not.
     pub async fn get_value(&self) -> anyhow::Result<String> {
-        if self.is_valid().await {
-            return match &self.data.read().await.value {
-                Some(value) => Ok(value.clone()),
-                None => Err(anyhow!("could no read the token's value")),
-            };
+        match self.is_valid().await {
+            true => {
+                let data_locked = self.data.read().await;
+                match &data_locked.value {
+                    Some(value) => Ok(value.clone()),
+                    None => Err(anyhow!("ERROR : Could not read the token's value")),
+                }
+            }
+            false => match self.refresh().await {
+                Ok(token) => Ok(token),
+                Err(e) => Err(anyhow!("ERROR : Failed to refresh token: {}", e)),
+            },
         }
-
-        self.refresh().await
```

Comment:
How is the new version better than the old one? If all you want is to add a message when `refresh` fails, use `context` or `with_context`: https://docs.rs/anyhow/latest/anyhow/